go/libraries/doltcore/sqle/dolt_map_iter.go

```diff
@@ -68,79 +107,83 @@ func (conv *KVToSqlRowConverter) ConvertKVToSqlRow(k, v types.Value) (sql.Row, e
 	}
 
 	cols := make([]interface{}, conv.rowSize)
-	filled, err := conv.processTuple(cols, 0, keyTup)
-	if err != nil {
-		return nil, err
+	if conv.valsFromKey > 0 {
+		// keys are not in sorted order so cannot use max tag to early exit
+		err := conv.processTuple(cols, conv.valsFromKey, 0xFFFFFFFFFFFFFFFF, keyTup)
+
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	if !valTup.Empty() {
-		filled, err = conv.processTuple(cols, filled, valTup)
+	if conv.valsFromVal > 0 {
+		err := conv.processTuple(cols, conv.valsFromVal, conv.maxValTag, valTup)
+
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	return cols, err
+	return cols, nil
 }
 
-func (conv *KVToSqlRowConverter) processTuple(cols []interface{}, filled int, tup types.Tuple) (int, error) {
+func (conv *KVToSqlRowConverter) processTuple(cols []interface{}, valsToFill int, maxTag uint64, tup types.Tuple) error {
 	tupItr, err := tup.Iterator()
 
 	if err != nil {
-		return 0, err
+		return err
 	}
 
-	for filled < len(conv.tagToSqlColIdx) {
+	filled := 0
+	var tag64 uint64
+	for filled < valsToFill && tag64 < maxTag {
 		_, tag, err := tupItr.Next()
 
 		if err != nil {
-			return 0, err
+			return err
 		}
 
 		if tag == nil {
 			break
 		}
 
-		if sqlColIdx, ok := conv.tagToSqlColIdx[uint64(tag.(types.Uint))]; !ok {
+		tag64 = uint64(tag.(types.Uint))
+
+		if tag64 > maxTag {
+			break
+		}
+
+		if sqlColIdx, ok := conv.tagToSqlColIdx[tag64]; !ok {
 			err = tupItr.Skip()
 
 			if err != nil {
-				return 0, err
+				return err
 			}
 		} else {
 			_, val, err := tupItr.Next()
 
 			if err != nil {
-				return 0, err
+				return err
 			}
 
 			cols[sqlColIdx], err = conv.cols[sqlColIdx].TypeInfo.ConvertNomsValueToValue(val)
 
 			if err != nil {
-				return 0, err
+				return err
 			}
 
 			filled++
 		}
 	}
 
-	return filled, nil
+	return nil
 }
 
 // KVGetFunc defines a function that returns a Key Value pair
 type KVGetFunc func(ctx context.Context) (types.Value, types.Value, error)
 
-// DoltMapIter uses a types.MapIterator to iterate over a types.Map and returns sql.Row instances that it reads and
-// converts
-type DoltMapIter struct {
-	kvGet KVGetFunc
-	conv  *KVToSqlRowConverter
-}
-
-// NewDoltMapIterFromNomsMapItr returns an iterator which returns sql.Row instances read from a types.Map.  The cols
-// passed in are used to limit the values that are processed
-func NewDoltMapIterFromNomsMapItr(mapItr types.MapIterator, cols []schema.Column) *DoltMapIter {
-	getFunc := func(ctx context.Context) (types.Value, types.Value, error) {
+func GetGetFuncForMapIter(mapItr types.MapIterator) func(ctx context.Context) (types.Value, types.Value, error) {
```

Comment:
not sure what you mean.  Do you just mean in general we should change the noms map iterators to return io.EOF instead of using a nil key to signify all items have been iterated over?